src/nvim/eval.c

```diff
@@ -10129,28 +10129,38 @@ static void f_getcompletion(typval_T *argvars, typval_T *rettv, FunPtr fptr)
   xpc.xp_pattern = get_tv_string(&argvars[0]);
   xpc.xp_pattern_len = STRLEN(xpc.xp_pattern);
   xpc.xp_context = cmdcomplete_str_to_type(get_tv_string(&argvars[1]));
-  if (xpc.xp_context == EXPAND_NOTHING) {
-    if (argvars[1].v_type == VAR_STRING) {
-      EMSG2(_(e_invarg2), argvars[1].vval.v_string);
-    } else {
-      EMSG(_(e_invarg));
-    }
-    return;
-  }
+  if (strcmp((char *)get_tv_string(&argvars[1]), "cmdline") == 0) {
```

Comment:
here `argvars[1]` is used  before `argvars[1].v_type == VAR_STRING` validates it.